packages/bp/src/core/messaging/messaging-service.ts

```diff
@@ -56,9 +54,17 @@ export class MessagingService {
       handler: this.handleOutgoingEvent.bind(this)
     })
 
-    await AppLifecycle.waitFor(AppLifecycleEvents.STUDIO_READY)
+    this.messaging = new MessagingChannel({
+      url: this.getMessagingUrl(),
+      adminKey: process.env.INTERNAL_PASSWORD,
+      axios: this.getAxiosConfig()
+    })
+    this.messaging.on('user', this.handleUserNewEvent.bind(this))
+    this.messaging.on('started', this.handleConversationStartedEvent.bind(this))
+    this.messaging.on('message', this.handleMessageNewEvent.bind(this))
 
-    this.clientSync = new MessagingClient({ url: this.getMessagingUrl(), config: this.getAxiosConfig() })
+    await AppLifecycle.waitFor(AppLifecycleEvents.STUDIO_READY)
+    this.messaging.url = this.getMessagingUrl()
```

Comment:
We don't actually know what the port of messaging is before this point so we have to set it after